test/unit/org/apache/cassandra/sensors/SensorsIndexWriteTest.java

```diff
@@ -0,0 +1,254 @@
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sensors;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.apache.cassandra.SchemaLoader;
+import org.apache.cassandra.cql3.ColumnIdentifier;
+import org.apache.cassandra.cql3.statements.schema.IndexTarget;
+import org.apache.cassandra.db.ColumnFamilyStore;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.db.Mutation;
+import org.apache.cassandra.db.MutationVerbHandler;
+import org.apache.cassandra.db.RowUpdateBuilder;
+import org.apache.cassandra.db.compaction.CompactionManager;
+import org.apache.cassandra.db.marshal.AsciiType;
+import org.apache.cassandra.dht.Murmur3Partitioner;
+import org.apache.cassandra.index.sai.StorageAttachedIndex;
+import org.apache.cassandra.io.util.DataOutputBuffer;
+import org.apache.cassandra.net.Message;
+import org.apache.cassandra.net.MessagingService;
+import org.apache.cassandra.net.SensorsCustomParams;
+import org.apache.cassandra.net.Verb;
+import org.apache.cassandra.schema.IndexMetadata;
+import org.apache.cassandra.schema.Indexes;
+import org.apache.cassandra.schema.KeyspaceParams;
+import org.apache.cassandra.utils.BloomFilter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SensorsIndexWriteTest
+{
+    public static final String KEYSPACE1 = "SensorsIndexWriteTest";
+    public static final String CF_STANDARD = "Standard";
+    public static final String CF_STANDARD_SAI = "StandardSAI";
+    public static final String CF_STANDARD_SECONDARY_INDEX = "StandardSecondaryIndex";
+
+    private CopyOnWriteArrayList<Message> capturedOutboundMessages;
+
+    @BeforeClass
+    public static void defineSchema() throws Exception
+    {
+        SchemaLoader.prepareServer();
+
+        // build SAI indexes
+        Indexes.Builder saiIndexes = Indexes.builder();
+        saiIndexes.add(IndexMetadata.fromSchemaMetadata(CF_STANDARD_SAI + "_val", IndexMetadata.Kind.CUSTOM, new HashMap<>()
+        {{
+            put(IndexTarget.CUSTOM_INDEX_OPTION_NAME, StorageAttachedIndex.class.getName());
+            put(IndexTarget.TARGET_OPTION_NAME, "val");
+        }}));
+
+        // build secondary indexes
+        Indexes.Builder secondaryIndexes = Indexes.builder();
+        secondaryIndexes.add(IndexMetadata.fromIndexTargets(Collections.singletonList(
+                                                            new IndexTarget(
+                                                            new ColumnIdentifier("val", true), IndexTarget.Type.VALUES)), CF_STANDARD_SECONDARY_INDEX + "_val",
+                                                            IndexMetadata.Kind.COMPOSITES,
+                                                            Collections.emptyMap()));
```

Comment:
Slightly more readable:
```suggestion
        IndexTarget indexTarget = new IndexTarget(new ColumnIdentifier("val", true), IndexTarget.Type.VALUES);
        secondaryIndexes.add(IndexMetadata.fromIndexTargets(Collections.singletonList(indexTarget), 
                                                            CF_STANDARD_SECONDARY_INDEX + "_val",
                                                            IndexMetadata.Kind.COMPOSITES,
                                                            Collections.emptyMap()));
```